### PR TITLE
Update server when video is in state `finished`

### LINKED ISSUE #1567: Returning from HDR at end of file fails
## Software Versions

- Jellyfin Server Version:  10.8.10
- Roku Client Version:  2.0 build 0

## Describe the bug
When playing any HDR file, either HDR10 or Dolby vision and pressing the back button on the remote control the video stops and the Roku returns to SDR mode. However if you let the file play to the end and the video naturally stops on it's own the Roku remains in HDR mode when returning to the interface.

This is using a Streaming Stick 4k model 3820EU and has been tested on two different TV's. Also the Roku's own media player handles this fine, the Roku returns to SDR mode regardless of how you stop the file, from the 'back' button or letting it play to the end.

## How To Reproduce

1. Play any HDR video
2. Press 'back' to stop the video and the Roku returns to SDR mode
3. Play any HDR video
4. Allow the video to stop naturally at the end and the Roku remains in HDR mode and doesn't return to SDR mode

## Expected behavior

Regardless of weather you manually stop a video of allow it to stop at the end the Roku should return to SDR mode.

## Connection Information

- Is server local or remote? Local
- Is server connection HTTP or HTTPS? HTTP

### LINKED ISSUE #1814: Roku client appears to NOT update NFO file after video finishes playing through to the very end
## Software Versions
- Jellyfin Server Version:  10.9.7
- Roku Client Version:  2.0.7

## Describe the bug
Upon allowing a video to play through to the very end, the Roku client appears to NOT prompt the update of the corresponding NFO file.

To be very specific, "watched" is NOT updating from "false" to "true" in the corresponding NFO file.

## Prerequisites To Reproduce
1. Navigate to the Libraries > Libraries section of the JellyFin server from the Dashboard in the main navigation
2. Select the "Manage library" option in the kebab menu for the testing library
3. Select the "Nfo" checkbox in the "Metadata savers" section for the testing library
4. Select "OK" to save the setting
5. Navigate to the Libraries > NFO Settings section of the JellyFin server in the main navigation menu
6. Select the the user to be used for testing in the "Save user watch data to NFO files for" field
7. Select "Save" to save the setting

## How To Reproduce
1. Open the JellyFin Roku client
2. Ensure you are logged in with the same user selected in Step 6 of the "Prerequisites To Reproduce" section
3. Navigate to the library selected in Step 2 of the "Prerequisites To Reproduce" section
4. Select any video that has not yet been watched
5. Watch the video through to the very end when the stream closes and returns you to the library

## Expected behavior
The NFO file is updated including "playcount" and "watched" among others when applicable

## Logs
There are no [ERR] messages in the logs for my testing time range. There was an instance of the following [WRN] message, but it does not appear to correlate with the behavior experienced.

`[2024-07-03 10:52:25.892 -04:00] [WRN] WS "10.16.8.1" error receiving data: "The remote party closed the WebSocket connection without completing the close handshake."`

## Connection Information
- Is server local or remote? **LOCAL**
- Is server connection HTTP or HTTPS? **HTTP**

## Additional context
If the video is exited by the user prior to conclusion, the NFO file is updated as expected whether that be the "watched", "playcount","lastplayed", or "resume" values.

Furthermore, the behavior is NOT reproducible from the Android or Web clients.

### PR DESCRIPTION
**Pull request by cewert**
## Changes
- Update server when video state = finished
- Update server when video state = buffering and m.playReported = false

## Issues
Fixes #1814

### CODE REVIEW

**Comment by cewert** [comment]
@roidy I added a commit to simulate hitting the back button when state is finished. Can you give that a test? Here's the new build https://github.com/jellyfin/jellyfin-roku/actions/runs/11087045525?pr=1941

File: components/PlaystateTask.bs [hunk 1]
```diff
@@ -10,7 +10,7 @@ sub PlaystateUpdate()
         url = "Sessions/Playing"
     else if m.top.status = "stop"
         url = "Sessions/Playing/Stopped"
-    else if m.top.status = "update"
+    else if m.top.status = "update" or m.top.status = "finished"
```
Comment (jimdogx) [inline components/PlaystateTask.bs]: I think you put the Finished on the wrong line :smile:  Shouldn't it go with "Stop"?

Comment (cewert) [reply components/PlaystateTask.bs]: Good catch!

**Comment by cewert** [comment]
@roidy the problem in your video should be fixed. I had the wrong api endpoint. https://github.com/jellyfin/jellyfin-roku/actions/runs/11087871463?pr=1941

**Review by jimdogx — APPROVED** [review]

File: components/video/VideoPlayerView.bs [hunk 2]
```diff
@@ -623,7 +637,6 @@ sub onState(msg)
 
         ' Stop playback and exit player
         m.top.control = "stop"
-        m.top.backPressed = true
```
Comment (cewert) [inline components/video/VideoPlayerView.bs:626]: `m.top.backPressed` wasn't being observed or used so I removed them

**Pull request merged**